Code/GraphMol/ForceFieldHelpers/CrystalFF/TorsionPreferences.cpp

```diff
@@ -0,0 +1,674 @@
+        "[nX2H0:1][c:2](c)!@;-[CX3:3]([!O])=[O:4] 1 66.5 1 0.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[nX2H0:1][a:2]([nX2H0])!@;-[CX3:3]=[O:4] 1 0.0 -1 36.8 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[*^2]!@;-[cH0:1][c:2]([cH1])!@;-[CX3:3]=[O:4] 1 0.0 -1 8.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH0:1][c:2]([cH1])!@;-[CX3:3]=[O:4] 1 0.0 -1 10.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH0:1][c:2]([cH0])!@;-[CX3:3]=[O:4] 1 0.0 1 0.0 1 0.0 -1 3.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([cH1])!@;-[CX3:3]([CX3H0])=[O:4] 1 0.0 -1 5.9 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([cH1])!@;-[CX3:3]=[O:4] 1 0.0 -1 5.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[a:1][a:2]!@;-[CX3:3]=[O:4] 1 0.0 -1 8.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([nX2])!@;-[CX3:3]=[NX3:4] 1 0.0 -1 8.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([nX3H1])!@;-[CX3:3]=[NX2:4] 1 97.4 1 0.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([nX2])!@;-[CX3:3]=[NX2:4] 1 0.0 -1 8.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([cH0]([OH1]))!@;-[CX3:3]=[NX2:4] 1 100.0 1 0.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([cH0])!@;-[CX3:3]=[NX2:4] 1 100.0 1 0.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH1:1][c:2]([cH1])!@;-[CX3:3]=[NX2:4] 1 0.0 -1 9.5 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[cH0:1][c:2]([cH0])!@;-[CX3:3]=[NX2:4] 1 0.0 1 -0.7 1 0.0 1 -0.5 1 0.0 1 -0.6\n"
+        "[a:1][c:2]!@;-[CX3:3]=[CX3H0:4] 1 0.0 -1 0.7 1 0.0 1 0.6 1 0.0 1 -0.5\n"
+        "[a:1][a:2]!@;-[CX3:3]=[CX3H2:4] 1 0.0 -1 1.3 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[a:1][a:2]!@;-[CX3:3]=[CX3H1:4] 1 0.0 -1 5.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[*:1][SX2:2]!@;-[SX2:3][*:4] 1 0.0 1 12.9 1 0.0 1 0.0 1 0.0 1 0.0\n"
+         // ------
+         //  these patterns were added as part of the ET-DG work
+         // non-aromatic double bonds
+         //"[*:1][CX3:2]=[CX3:3][*:4] 1 0.0 -1 7.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+        "[*:1][X3,X2:2]=[X3,X2:3][*:4] 1 0.0 -1 7.0 1 0.0 1 0.0 1 0.0 1 0.0\n"
+    ;
+
+    //! A structure used to the experimental torsion patterns
+    struct ExpTorsionAngle {
+      std::string smarts;
+      std::vector<double> V;
+      std::vector<int> signs;
+      boost::shared_ptr<const ROMol> dp_pattern;
+      unsigned int idx[4];
+    };
+
+    // class to store the experimental torsion angles
+    class ExpTorsionAngleCollection {
+      public:
+        typedef std::vector<ExpTorsionAngle> ParamsVect;
+        static const ExpTorsionAngleCollection *getParams(const std::string &paramData="");
+        ParamsVect::const_iterator begin() const { return d_params.begin(); };
+        ParamsVect::const_iterator end() const { return d_params.end(); };
+        ExpTorsionAngleCollection(const std::string &paramData);
+      private:
+        ParamsVect d_params;                                 //!< the parameters
+    };
+
+    typedef boost::flyweight<boost::flyweights::key_value<std::string,ExpTorsionAngleCollection>,
```

Comment:
The use of the flyweights here indicates that you've thought about multithreading and this code. Have you actually tested it with multiple threads?